### PR TITLE
Improved signupform validations

### LINKED ISSUE #696: Password Validations are not enforced across the app.
**The bug**
In case of registering a user on the portal, there is a constraint imposed saying that the password should be 8 characters long, but no such checks are being taken care of at the time of creating the passwords or on the **Forget Password** pages.
Passwords of length smaller are also being allowed. Further Only strong passwords should be allowed on our App for better security.

**Expected behavior**
After the solving of this issue, only the passwords adhering to the constraints will be allowed during new user registration and on the forgot password page.

**Screenshot**
WhatsApp Image 2023-03-14 at 3 05 04 PM
WhatsApp Image 2023-03-14 at 3 20 58 PM

I will happily take up this issue, if approved.

### PR DESCRIPTION
**Pull request by jmalisha**
Improved Sign Up form validation by creating a new validation function making it modular and easy to extend.
Validation breakdown:
- First name: Should be at least two characters
- Last name: Should be at least two characters
- Email: Should be a valid email (based on the defined regex)
- Password: Must be at least 8 characters, and must contain at least a letter,  number, and special character.

**What kind of change does this PR introduce?**

A bugfix

**Issue Number:**

- Fixes #696

**Did you add tests for your changes?**

No
**Snapshots/Videos:**
Here is a snapshot

https://user-images.githubusercontent.com/51859049/229231308-e4ab20be-b662-4510-9448-6b7a30a7a304.mp4

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

File: src/screens/LoginPage/LoginPage.tsx [hunk 1]
```diff
@@ -116,11 +116,82 @@ function LoginPage(): JSX.Element {
...
   };
 
+  interface SignupFormData {
+    signfirstName: string;
+    signlastName: string;
+    signEmail: string;
+    signPassword: string;
+    cPassword: string;
+  }
+  interface ValidationResult {
```
Comment (Rumone) [inline src/screens/LoginPage/LoginPage.tsx:126]: Why not use a type here instead of interfaces

**Comment by palisadoes** [comment]
I have to close this PR. The expectation was that the markdown files would be adjusted so that the H1, H2, H3, H4 titles would be appropriately assigned to the headings to make the table of contents usable.

We are not accepting any more changes to CSS.

**Comment by palisadoes** [comment]
Sorry, wrong PR

**Review by Rumone — CHANGES_REQUESTED** [review]

**Comment by EshaanAgg** [comment]
@jmalisha 
Instead of issuing one error at a time, you can consider pushing all the errors to an array and then showing one toast for each error object.

This way, lets say I have two strong fields, `firstName` and `lastName`, then both of them would be validated in one request and I would be shown two toasts asking me to fix each of them. With the current implementation, I would first submit the form to find that my `firstName` is invalid, and then I would need to submit it again to find that my `lastName` is invalid! Using an array of errors is just a better user experience, in my opinion.

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

**Comment by aashimawadhwa** [comment]
@jmalisha   please take a pull the failing test will pass.

**Comment by palisadoes** [comment]
@jmalisha Please merge with the upstream and then push the latest changes to this PR.

**Comment by jmalisha** [comment]
@palisadoes Please have a look at it. Thanks

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by jmalisha** [comment]
@palisadoes Could you please have a look at it again

**Review by noman2002 — APPROVED** [review]
Looks good to me.

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
Please fix the conflicting files

**Review by Kevoniat — APPROVED** [review]

**Comment by palisadoes** [comment]
Please fix the conflicting files.

**Comment by jmalisha** [comment]
@palisadoes working on it

File: src/screens/LoginPage/LoginPage.tsx (hunk 1 above)
Comment (palisadoes) [reply src/screens/LoginPage/LoginPage.tsx]: @Rumone is this resolved?

**Pull request merged**